runtime/vm/UpcallVMHelpers.cpp

```diff
@@ -227,28 +227,44 @@ getInternalTypeFromSignature(J9JavaVM *vm, J9Class *typeClass, U_8 sigType)
  *
  * @param argType the predefined signature type
  * @param argValue the requested argument value
- * @return a U_32 value for the normalized argument
+ * @return a I_32 value for the normalized argument
  *
  * Note:
- * The argument normalization is only intended for the type less then 4 bytes
+ * The argument normalization is only intended for types less than 4 bytes
  * in java, which include boolean (1 byte), byte, char (2 bytes), short.
  */
-static U_32
+static I_32
 getNormalizedArgValue(U_8 argType, I_32 argValue)
 {
 	I_32 realValue = argValue;
 
 	switch (argType) {
 	case J9NtcBoolean:
-		realValue = ((realValue & J9_FFI_UPCALL_BYTE_TYPE_MASK) > 0) ? 0x1 : 0x0;
+	{
+		static const I_32 boolTypeMask = 0xFF;
+		realValue = J9_ARE_ANY_BITS_SET(realValue, boolTypeMask) ? 0x1 : 0x0;
```

Comment:
Use a macro if you want to the constant to have a name; we want neither the space required for the static nor the memory access. Same comment for `charTypeMask`.

Reply:
Updated the code by directly using `0xFF` and `0xFFFF` based on the offline discussion.